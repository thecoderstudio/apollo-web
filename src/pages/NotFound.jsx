import React from 'react';
import styled from 'styled-components';
import media from '../util/media';
import Link from '../components/links/Link';
import OutlinedButton from '../components/buttons/OutlinedButton';
import notFoundImg from '../images/lost.svg';

const Container = styled.div`
  position: fixed;
  left: 50%;
  transform: translate(-50%, 0);
  width: 100%;
  max-width: 1750px;
  height: 100%;
  display: grid;
  grid-template-columns: [img] 1fr [content] 1fr;
  justify-items: center;
  align-items: center;

  ${
    media.phone`
      grid-template-columns: [content] 1fr;
    `
  }
`;

const SupportingImg = styled.img`
  grid-column: img;
  width: 75%;

  ${
    media.phone`
      display: none;
    `
  }
`;

const Content = styled.div`
  display: flex;
  grid-column: content;
  flex-direction: column;
  height: 100%;
  justify-content: center;
  align-items: center;
`;

const Title = styled.h1`
  margin: 0px;
  text-align: center;
  font-size: 20em;
  color: ${props => props.theme.primary};

  ${
    media.phone`
      font-size: 10em;
    `
  }
`;

const SubTitle = styled.h2`
  margin: 0px;
  text-align: center;
  font-size: 2em;
  color: ${props => props.theme.darkGrey};

  ${
    media.phone`
      font-size: 1em;
    `
  }
`;

const StyledButton = styled(OutlinedButton)`
  margin-top: 64px;
  border: 1px solid ${props => props.theme.primary};
`;

export default function NotFound(props) {
  return (
    <Container>
      <SupportingImg src={notFoundImg} />
      <Content>
        <Title>404</Title>
        <SubTitle>Houston, we have a problem</SubTitle>
        <StyledButton><Link to='/'>Reroute</Link></StyledButton>
      </Content>
    </Container>
  );
}
