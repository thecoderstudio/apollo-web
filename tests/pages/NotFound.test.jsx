import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import NotFound from '../../src/pages/NotFound';

test("not found renders correctly", () => {
  const tree = renderer.create(
    <BrowserRouter>
      <NotFound />
    </BrowserRouter>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
