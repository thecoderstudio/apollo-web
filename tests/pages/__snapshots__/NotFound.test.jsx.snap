// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`not found renders correctly 1`] = `
.c6 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c6:hover {
  opacity: 0.90;
}

.c0 {
  position: fixed;
  left: 50%;
  -webkit-transform: translate(-50%,0);
  -ms-transform: translate(-50%,0);
  transform: translate(-50%,0);
  width: 100%;
  max-width: 1750px;
  height: 100%;
  display: grid;
  grid-template-columns: [img] 1fr [content] 1fr;
  justify-items: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  grid-column: img;
  width: 75%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  grid-column: content;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  margin: 0px;
  text-align: center;
  font-size: 20em;
}

.c4 {
  margin: 0px;
  text-align: center;
  font-size: 2em;
}

.c5 {
  max-height: 50px;
  padding: 12px;
  border: none;
  border-radius: 5px;
  color: white;
  font-family: 'B612',sans-serif;
  font-size: 1rem;
  font-weight: 700;
  cursor: pointer;
  background-color: transparent;
  border: 1px solid;
  margin-top: 64px;
  border: 1px solid;
}

.c5:hover {
  opacity: 0.90;
}

@media (max-width:47.9375em) {
  .c0 {
    grid-template-columns: [content] 1fr;
  }
}

@media (max-width:47.9375em) {
  .c1 {
    display: none;
  }
}

@media (max-width:47.9375em) {
  .c3 {
    font-size: 10em;
  }
}

@media (max-width:47.9375em) {
  .c4 {
    font-size: 1em;
  }
}

<div
  className="c0"
>
  <img
    className="c1"
    src={[Function]}
  />
  <div
    className="c2"
  >
    <h1
      className="c3"
    >
      404
    </h1>
    <h2
      className="c4"
    >
      Houston, we have a problem
    </h2>
    <button
      className="sc-AxirZ sc-AxiKw c5"
    >
      <a
        className="c6"
        href="/"
        onClick={[Function]}
      >
        Reroute
      </a>
    </button>
  </div>
</div>
`;
